### PR TITLE
Improve downloads

### PR DESCRIPTION
**Pull request by MassiveBox**
Hello!  

I'm back with another PR that fixes a bug caused by the latest release of Lokinet (the software, not the website): Since it didn't contain an .exe file, the download page for Windows doesn't currently work on lokinet.org, because it's coded to return a 404 error if it doesn't find the build we want.  
To address this issue, in my PR I made it so if we don't find the file we want in the latest release, we'll get the list of releases via GitHub's API, and scan them from newest to oldest (skipping the first one of course), to find the newest with the release for the OS we want.  

We could have coded it to fetch all the releases and start scanning from the first one, but I think this is the best way, because it's more common that there are both an exe and tarball in the same release, and in that case the server only does an API call; the tradeoff being that if there's no desired file in the latest release, we'll have to perform at least 3 calls instead of only two. But again this should be more rare.

As always let me know if something's wrong or could be done better. @yougotwill keep the nitpicking coming ;D

### CODE REVIEW

**Comment by yougotwill** [comment]
@MassiveBox This just got brought to my attention and was just about to fix it 😂

File: pages/linux.tsx [hunk 1]
```diff
@@ -7,40 +8,50 @@ function Page() {
...
-    return {
-      notFound: true,
-    }
-  }
-  const assetsUrl =  json.assets.map((m:any) => m.browser_download_url)
-  if (!assetsUrl || !assetsUrl.length) {
-    return {
-      notFound: true,
-    }
+  var downloadURL = await GetDownloadURL("latest", ".tar.xz");
```
Comment (majestrate) [inline pages/linux.tsx:11]: we have linux packages repos at deb.oxen.io and rpm.oxen.io
i'd suggest making a new page with instructions on how to use both as the github releases are just signed source code tarballs.

**Comment by yougotwill** [comment]
@MassiveBox Since this needs to get fixed fast I'm going to take your code and do my nitpicking and merge. But I'll make sure to explain the changes in this PR and credit you in the merge. 💪 

Thank you very much for another PR! I was talking at the office about the importance of making our different repo's more contributor friendly and you have just proven that investing the time pays off. My sincere appreciation 🙏

**Comment by MassiveBox** [comment]
No problem! I'm here to help, not slow things down :P  
Now I can see Majestrate's inline comments btw, I'm glad that got sorted out. If you decide to make some kind of Linux page (or even a popup would suffice, since it's not much information), I could help with that. Let me know if my contribution would be helpful, or if you prefer keeping styling and designing the base website to the professionals, it's fine nonetheless.

**Comment by yougotwill** [comment]
@MassiveBox All good! Tell you what. I'll put this quick fix up just so that the links are working. If you like have a go at creating a new page for downloads with better instructions and links as suggested by @majestrate. This should be done in a separate PR.

No pressure or deadline but keep it professional. 😉

**Comment by yougotwill** [comment]
@MassiveBox I've made the merge [here](https://github.com/oxen-io/lokinet-website/pull/9) and the changes will be reflected soon. I have to hop onto some other tasks for now but I'll leave this PR open as a reminder and will give you more detailed feedback when I can.

File: pages/api/download/[platform].ts [hunk 2]
```diff
@@ -0,0 +1,87 @@
+import { NextApiRequest, NextApiResponse } from "next";
```
Comment (majestrate) [inline pages/api/download/[platform].ts:1]: are square braces some special syntax?

Comment (majestrate) [reply pages/api/download/[platform].ts:1]: square braces in the file name rather.

Comment (yougotwill) [reply pages/api/download/[platform].ts:1]: @majestrate Yeah it's a special syntax for NextJS. https://nextjs.org/docs/routing/dynamic-routes

**Comment by yougotwill** [comment]
@MassiveBox My apologies I think I push to your fork by mistake. That's what I get for using a git GUI and not the commandline.

**Comment by MassiveBox** [comment]
Looks like I forgot to close this, doing it now x.x

I'll take this opportunity to ask @yougotwill if I can send you an email to discuss about a (quite drastic) solution to the problems with GitBook docs I've highlighted in [this](https://github.com/oxen-io/oxen-docs-gitbook/issues/6) issue, that also opens many new opportunities. Yup, it involves migrating away from GitBook to a self hosted static site, but the transition in my prototype is very easy, and to me, the pros outweigh the cons heavily.  
I'm asking because I know you're all busy working on other projects for the OPTF, and I absolutely don't want to be the annoying kid that adds work on top of other (probably more important) work.

Please let me know :)

**Comment by yougotwill** [comment]
@MassiveBox Let me chat with some people before we take this discussion further. I'll get back to you ASAP. However, taking a look at your issue I think its has been resolved and we just forgot to close the issue? Just checked on Firefox iOS.

Can you elaborate more about your prototype here please. 😄

**Comment by MassiveBox** [comment]
Firefox on iOS is entirely different from Firefox android, you can think of it just a reskinned Safari with Mozilla branding and a few changes: that's how browsers on iOS work, you can read more at 2.5.6 [here](https://developer.apple.com/app-store/review/guidelines/).

And even if the glitched layout is fixed, there's still a giant privacy problem: GitBook pings Firebase, Firestore and Google Analytics (!), on top of iFrame, Polyfill, Sentry and Amplitude. It's bad advertisement for a project that promotes privacy.

My proposal is to build a static documentation site based on Codeberg Docs ([web](https://docs.codeberg.org/) - [source](https://codeberg.org/Codeberg/Documentation)). It's already feature rich, heavily customizable, and needs very little work to match the current style and to get the current features. I've stared working on a couple things, here's a quick demo:

https://user-images.githubusercontent.com/71317968/143073579-1ec0bfed-a607-4ef6-b561-792d864b7d15.mp4

So far I haven't ported everything to the new menu system, which is the only thing that's a bit different from the current docs. That's why there are only an handful of items in the menu.

You can see I've added drop down menus (that work even without JavaScript), I can confirm editing the default layout is a piece of cake.

The major problem I could see is the disadvantage of having to self host, however this website is not heavy, it's made with Eleventry which is a well established static site generator.

Please let me know what you think of this! I haven't spent too much effort on this yet because I don't know whether it will be approved or not, however I'm very confident it can become much more beautiful with little effort.

**Pull request closed without merging**